### PR TITLE
Absent on lwp

### PR DESCRIPTION
**Pull request by ashish-greycube**
Solution to https://github.com/frappe/hrms/issues/193 and
https://github.com/frappe/erpnext/issues/5893

### CODE REVIEW

**Comment by jigneshpshah** [comment]
Design : 

in HR Settings

There will be options for **_Salary Slip Based On_**
Leave Application
Attendance
Timesheet

if user will choose _Attendance_ , he will get an option to _Consider Absent as LWP_ or not, based on which Absent will be treated as LWP during Payroll processing.

Below is the gif explaining a scenario of an Employee having attendance as below
Working Days = 20

Absent Days = 3
Leave without Application = 1
1 Half Day = 0.5
Hence Total LWP = 4.5

Payment Days = 15.5

salaryslip_absentaslwp

File: erpnext/hr/doctype/salary_slip/salary_slip.py [hunk 1]
```diff
@@ -374,24 +374,71 @@ def get_holidays_for_employee(self, start_date, end_date):
...
+					and leave_type is NULL
+					and employee = %(employee)s
+					and attendance_date between %(st_dt)s and %(end_dt)s""",
+					{"employee": self.employee, "st_dt": self.start_date,"end_dt": self.end_date})[0][0]
+
+
+		half_day_count = frappe.db.sql("""
+				select count(status) 
+				from `tabAttendance` as t1 
+                where t1.docstatus = 1
```
Comment (rmehta) [inline erpnext/hr/doctype/salary_slip/salary_slip.py]: inconsistent indenting

**Review by rmehta — CHANGES_REQUESTED** [review]
please add tests

@ashish-greycube Thanks for contributing! Let me know if you need a call regarding tests.

**Comment by rmehta** [comment]
Closing for now to reduce PR count. Please re-open after tests.

**Comment by govindsmenokee** [comment]
@rmehta I think a better approach would be to move the attendance required check to the employee doc level, that way we have more control over which set of employees should have attendance. Also having this check at any other place would not make sense with the current system.

**Comment by Olawale1** [comment]
Hi @ashish-greycube 

Trust you're doing well. Any chance of adding some tests so this can go through? It's still a much needed feature and quite frankly, it's surprising ERPNext doesn't have any link between attendance and payroll up till now!

Thanks

**Comment by flexy2ky** [comment]
This will indeed be a brilliant addition to the Salary Module, whichever way it is implemented. We need this to reduce the number of processes required to confirm days worked. Right now, there are too many manual steps needed for organizations with a number of staff who may miss work without authorization, manually processing LWP for every single absence is a pain.

**Comment by amutaher** [comment]
> Design :
> 
> in HR Settings
> 
> There will be options for **_Salary Slip Based On_**
> Leave Application
> Attendance
> Timesheet
> 
> if user will choose _Attendance_ , he will get an option to _Consider Absent as LWP_ or not, based on which Absent will be treated as LWP during Payroll processing.
> 
> Below is the gif explaining a scenario of an Employee having attendance as below
> Working Days = 20
> 
> Absent Days = 3
> Leave without Application = 1
> 1 Half Day = 0.5
> Hence Total LWP = 4.5
> 
> Payment Days = 15.5
> 
> salaryslip_absentaslwp

@jigneshpshah 

that's exactly what we need what happens for this feature?

**Comment by Olawale1** [comment]
Any update on this please?

**Pull request closed without merging**